internal/services/mysql/mysql_flexible_server_resource.go

```diff
@@ -362,6 +362,9 @@ func resourceMysqlFlexibleServerCreate(d *pluginsdk.ResourceData, meta interface
 		return fmt.Errorf("waiting for creation of %s: %+v", id, err)
 	}
 
+	// Take some time for resource to be visible after creating. Hence, sleep for 30 seconds until issue https://github.com/Azure/azure-rest-api-specs/issues/21178 is fixed.
+	time.Sleep(30 * time.Second)
```

Comment:
we should never use sleeps to account for eventually consistency issues - while it could be 30 seconds today for you it could end up being 60 for someone else another day - instead can you add a state wait function that calls the API till it is visible?

Reply:
Thanks for your feedback. I have fixed the suggestion. Could you please take another look?